### PR TITLE
Apply changes for macro support on Windows

### PR DESCRIPTION
**Pull request by compnerd**

### CODE REVIEW

**Comment by compnerd** [comment]
@swift-ci please test

File: llvm/include/llvm/Support/DynamicLibrary.h [hunk 1]
```diff
@@ -48,6 +48,9 @@ class DynamicLibrary {
   /// Returns true if the object refers to a valid library.
   bool isValid() const { return Data != &Invalid; }
 
+  /// Return the OS specific handle value.
+  void *getOSSpecificHandle() const { return Data; }
```
Comment (etcwilde) [inline llvm/include/llvm/Support/DynamicLibrary.h]: This function already exists 5 lines up.

File: lldb/cmake/modules/AddLLDB.cmake [hunk 2]
```diff
@@ -217,17 +217,22 @@ function(add_properties_for_swift_modules target reldir)
       # Workaround for a linker crash related to autolinking: rdar://77839981
       set_property(TARGET ${target} APPEND_STRING PROPERTY
                    LINK_FLAGS " -lobjc ")
+
+    set_property(TARGET ${target} APPEND PROPERTY BUILD_RPATH "${SWIFT_BUILD_RPATH}")
```
Comment (etcwilde) [inline lldb/cmake/modules/AddLLDB.cmake]: Not a huge fan of the lack of indentation here.

**Comment by compnerd** [comment]
@swift-ci please test

**Comment by bnbarham** [comment]
This is missing the arch cherry-pick again @compnerd (https://github.com/apple/llvm-project/pull/7404/commits/d013aae1eba33246dc9a5dc4f61273442b7dfa54)

**Review by bnbarham — APPROVED** [review]

**Review by etcwilde — APPROVED** [review]

**Comment by bulbazord** [comment]
@compnerd  can you also cherry-pick this to `next`? Thanks!

**Pull request merged**